src/index.js

```diff
@@ -1,132 +1,255 @@
-/* global mw */
 import mux from 'mux-embed';
-// const log = mux.log;
-const assign = mux.utils.assign;
-const secondsToMs = mux.utils.secondsToMs;
-
-const muxPlugin = {
-  defaultConfig: {
-    options: {}
-  },
-
-  setup: function () {
-    const player = this.getPlayer();
-    const options = this.getConfig('options') || {data: {}};
-
-    options.data = assign({}, options.data, {
-      player_software_name: 'Kaltura',
-      player_software_version: player.evaluate('{playerVersion}'),
-      player_mux_plugin_name: 'kaltura-mux',
-      player_mux_plugin_version: '[AIV]{version}[/AIV]'
-    });
 
-    const playerId = player.id;
+const log = mux.log;
+const secondsToMs = mux.utils.secondsToMs;
+const assign = mux.utils.assign;
+const getComputedStyle = mux.utils.getComputedStyle;
+const safeCall = mux.utils.safeCall;
 
-    player.mux = {};
-    player.mux.emit = function (eventType, data) {
-      mux.emit(playerId, eventType, data);
-    };
+// eslint-disable-next-line no-undef
+const kalturaPlayer = KalturaPlayer;
 
-    options.getPlayheadTime = function () {
-      secondsToMs(player.evaluate('{video.player.currentTime}'));
-    };
+const initKalturaMux = function (player, options) {
```

Comment:
This assumes `KalturaPlayer` is on the window, correct? The pattern we use elsewhere is
```
const initKalturaMux = function (player, options, kaltura = window.KalturaPlayer) {
```
That pulls in the global instance by default, but also let's the caller specify a kaltura instance, if they're using npm package for example. This may all change with the kaltura plugin method, but just fyi